mathjax3-ts/output/chtml/Wrappers/msubsup.ts

```diff
@@ -0,0 +1,183 @@
+import {StyleList} from '../CssStyles.js';
+
+/*****************************************************************/
+/*
+ *  The CHTMLmsub wrapper for the MmlMsub object
+ */
+
+export class CHTMLmsub extends CHTMLscriptbase {
+    public static kind = MmlMsub.prototype.kind;
+
+    /*
+     * @override
+     */
+    public get script() {
+        return this.childNodes[(this.node as MmlMsub).sub];
+    }
+
+    /*
+     * Get the shift for the subscript
+     *
+     * @override
+     */
+    protected getOffset(bbox: BBox, sbox: BBox) {
+        return -this.getV(bbox, sbox);
+    }
+
+}
+
+/*****************************************************************/
+/*
+ *  The CHTMLmsup wrapper for the MmlMsup object
+ */
+
+export class CHTMLmsup extends CHTMLscriptbase {
+    public static kind = MmlMsup.prototype.kind;
+
+    /*
+     * @override
+     */
+    public get script() {
+        return this.childNodes[(this.node as MmlMsup).sup];
+    }
+
+    /*
+     * Get the shift for the superscript
+     *
+     * @override
+     */
+    public getOffset(bbox: BBox, sbox: BBox) {
+        return this.getU(bbox, sbox);
+    }
+
+}
+
+/*****************************************************************/
+/*
+ *  The CHTMLmsubsup wrapper for the MmlMsubsup object
+ */
+
+export class CHTMLmsubsup extends CHTMLscriptbase {
+    public static kind = MmlMsubsup.prototype.kind;
+
+    public static styles: StyleList = {
+        'mjx-script': {
+            display: 'inline-block',
+            'padding-right': '.05em'   // scriptspace
+        },
+        'mjx-script > *': {
+            display: 'block'
+        }
+    };
+
+    /*
+     *  Cached values for the script offsets and separation (so if they are
+     *  computed in computeBBox(), they don't have to be recomputed for toCHTML())
+     */
+    protected UVQ: number[] = null;
+
+    /*
+     * @return{CHTMLWrapper}  The wrapper for the subscript
+     */
+    public get sub() {
+        return this.childNodes[(this.node as MmlMsubsup).sub];
+    }
+
+    /*
+     * @return{CHTMLWrapper}  The wrapper for the superscript
+     */
+    public get sup() {
+        return this.childNodes[(this.node as MmlMsubsup).sup];
+    }
+
+    /*
+     * @override
+     */
+    public toCHTML(parent: HTMLElement) {
+        this.chtml = this.standardCHTMLnode(parent);
+        const [u, v, q] = this.getUVQ(this.base.getBBox(), this.sub.getBBox(), this.sup.getBBox());
+        const style = {'vertical-align': this.em(v)};
+        this.base.toCHTML(this.chtml);
+        const stack = this.chtml.appendChild(this.html('mjx-script', {style}));
+        this.sup.toCHTML(stack);
+        stack.appendChild(this.html('mjx-spacer', {style: {'margin-top': this.em(q)}}));
+        this.sub.toCHTML(stack);
+    }
+
+    /*
+     * @override
+     */
+    public computeBBox(bbox: BBox) {
+        const basebox = this.base.getBBox();
+        const subbox  = this.sub.getBBox();
+        const supbox  = this.sup.getBBox();
```

Comment:
ditto

Reply:
OK, changed to `baseChild`, `supChild`, `subChild`.